hibernate-core/src/main/java/org/hibernate/query/sqm/sql/internal/EntityValuedPathInterpretation.java

```diff
@@ -284,10 +292,28 @@ public static <T> EntityValuedPathInterpretation<T> from(
 			EntityValuedModelPart mapping,
 			EntityValuedModelPart treatedMapping,
 			SqmToSqlAstConverter sqlAstCreationState) {
+		final boolean expandToAllColumns;
+		final Clause currentClause = sqlAstCreationState.getCurrentClauseStack().getCurrent();
+		if ( currentClause == Clause.GROUP || currentClause == Clause.ORDER ) {
+			final SqmQuerySpec<?> querySpec = (SqmQuerySpec<?>) sqlAstCreationState.getCurrentSqmQueryPart();
```

Comment:
I'd prefer if we could pass this as argument to the `from` method instead of adding `getCurrentSqmQueryPart`

Reply:
This was needed because this `from` method can get called from different places for e.g. implicit joins, and I didn't want to add a parameter to all methods when it's only used for this specific check.